pkg/rehearse/jobs_test.go

```diff
@@ -1454,3 +1454,169 @@ func TestDetermineJobURLPrefix(t *testing.T) {
 		})
 	}
 }
+
+func TestSortConfigs(t *testing.T) {
+	testCases := []struct {
+		name     string
+		one      *config.DataWithInfo
+		two      *config.DataWithInfo
+		expected bool
+	}{
+		{
+			name: "same org/repo, branches main and release-4.10",
+			one: &config.DataWithInfo{
+				Info: config.Info{
+					Filename: "targetOrg-targetRepo-main.yaml",
+					Metadata: api.Metadata{
+						Org:    "targetOrg",
+						Repo:   "targetRepo",
+						Branch: "main",
+					},
+				},
+			},
+			two: &config.DataWithInfo{
+				Info: config.Info{
+					Filename: "targetOrg-targetRepo-release-4.10.yaml",
+					Metadata: api.Metadata{
+						Org:    "targetOrg",
+						Repo:   "targetRepo",
+						Branch: "release-4.10",
+					},
+				},
+			},
+			expected: true,
+		},
+		{
+			name: "different org/repo, branches main and release-4.10",
+			one: &config.DataWithInfo{
+				Info: config.Info{
+					Filename: "targetOrg-targetRepo-main.yaml",
+					Metadata: api.Metadata{
+						Org:    "targetOrg",
+						Repo:   "targetRepo",
+						Branch: "main",
+					},
+				},
+			},
+			two: &config.DataWithInfo{
+				Info: config.Info{
+					Filename: "anotherOrg-anotherRepo-release-4.10.yaml",
+					Metadata: api.Metadata{
+						Org:    "anotherOrg",
+						Repo:   "anotherRepo",
+						Branch: "release-4.10",
+					},
+				},
+			},
+			expected: true,
+		},
+		{
+			name: "same org/repo, branches release-4.9 and release-4.10",
+			one: &config.DataWithInfo{
+				Info: config.Info{
+					Filename: "targetOrg-targetRepo-release-4.9.yaml",
+					Metadata: api.Metadata{
+						Org:    "targetOrg",
+						Repo:   "targetRepo",
+						Branch: "release-4.9",
+					},
+				},
+			},
+			two: &config.DataWithInfo{
+				Info: config.Info{
+					Filename: "targetOrg-targetRepo-release-4.10.yaml",
+					Metadata: api.Metadata{
+						Org:    "targetOrg",
+						Repo:   "targetRepo",
+						Branch: "release-4.10",
+					},
+				},
+			},
+			expected: false,
```

Comment:
Ditto (this test nicely exhibits the actual bug in the current implementation - the one that you will fix)